lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.ts

```diff
@@ -280,11 +341,101 @@ export class ContentMetadataComponent implements OnChanges, OnInit, OnDestroy {
         this.categoryControlVisible = false;
     }
 
-    cancelChanges() {
+    cancelChanges(buttonType: ButtonType, event: MouseEvent) {
+        event.stopPropagation();
+        this.toggleEditMode(buttonType);
         this.revertChanges();
         this.loadProperties(this.node);
     }
 
+    isEditingPanel(): boolean {
+        return (
+            (this.editable && this.hasMetadataChanged) ||
+            (this.editableTags && this.hasMetadataChanged) ||
+            (this.editableCategories && this.hasMetadataChanged) ||
+            (this.editableGroup?.editable && this.hasMetadataChanged)
+        );
+    }
+
+    showSnackbar(message: string): void {
+        this.notificationService.showError(message);
+    }
+
+    toggleEdit(event: MouseEvent, group: CardViewGroup, buttonType: ButtonType): void {
+        if (this.isEditingPanel()) {
+            this.showSnackbar('METADATA.BASIC.SNACKBAR_MESSAGE');
+            return;
+        }
+
+        if (this.editableGroup && this.editableGroup !== group) {
+            this.editableGroup.editable = false;
+        }
+
+        event.stopPropagation();
+
+        switch (buttonType) {
+            case ButtonType.GeneralInfo:
+                this.editable = !this.editable;
+                this.editableChange.emit(this.editable);
+                this.panel.open();
+                this.editableTags = false;
+                this.editableCategories = false;
+                break;
+
+            case ButtonType.Tags:
+                this.editableTags = !this.editableTags;
+                this.editableTagsChange.emit(this.editableTags);
+                this.tagsPanelState = this.editableTags;
+                this.tagNameControlVisible = true;
+                break;
+
+            case ButtonType.Categories:
+                this.editableCategories = !this.editableCategories;
+                this.editableCategoriesChange.emit(this.editableCategories);
+                this.categoriesPanelState = this.editableCategories;
+                this.categoryControlVisible = true;
+                break;
+
+            case ButtonType.Group:
+                group.editable = !group.editable;
+                this.editableGroup = group.editable ? group : null;
+                if (group.editable) {
+                    group.expanded = true;
+                }
+                break;
+
+            default:
+                break;
+        }
+
+        if (buttonType !== ButtonType.GeneralInfo) {
+            this.editable = false;
+        }
+
+        if (buttonType !== ButtonType.Tags) {
+            this.editableTags = false;
+        }
+
+        if (buttonType !== ButtonType.Categories) {
+            this.editableCategories = false;
+        }
+    }
+
+    handleGeneralInfoPanelState() {
```

Comment:
confusing name as you don't handle the state but toggle it... rename to `toggleGeneralInfoPanel`